server/auctionapp/src/main/java/com/atlantbh/auctionapp/service/AuthService.java

```diff
@@ -98,4 +117,65 @@ public void authenticate(String email, String password) throws Exception {
         }
     }
 
+    public String updateResetPasswordToken(String email) throws MessagingException, UnsupportedEncodingException {
+        UserEntity userEntity = userRepository.findByEmail(email);
+        String token = RandomString.make(30);
+        String url = allowedURL + "/reset-password/" + token;
+        if (userEntity != null) {
+            userEntity.setResetPasswordToken(token);
+            userEntity.setResetPasswordTokenCreatedAt(LocalDateTime.now());
+            userRepository.save(userEntity);
+            sendEmail(email, url);
+        } else {
+            logger.error("User with email: " + email + " does not exist.");
+            throw new NotFoundException("User with email: " + email + " does not exist.");
+        }
+
+        return "Please check your email for the reset password link.";
```

Comment:
Backend shod not drive UI messages. We should have check on UI if thisRequest.statusCode === 200 then display this message